sdl_android/src/main/java/com/smartdevicelink/protocol/WiProProtocol.java

```diff
@@ -567,19 +618,43 @@ else if (frameInfo == FrameDataControlFrameType.StartSession.getValue()) {
 							return;
 						}
 
+						Log.d(TAG, "Checking secondary transport details");
+						ArrayList<String> secondary = (ArrayList<String>) packet.getTag(ControlFrameTags.RPC.StartServiceACK.SECONDARY_TRANSPORTS);
+						ArrayList<Integer> audio = (ArrayList<Integer>) packet.getTag(ControlFrameTags.RPC.StartServiceACK.AUDIO_SERVICE_TRANSPORTS);
+						ArrayList<Integer> video = (ArrayList<Integer>) packet.getTag(ControlFrameTags.RPC.StartServiceACK.VIDEO_SERVICE_TRANSPORTS);
+
+						if(secondary != null){
+							for(String s : secondary){
+								Log.d(TAG, "Supported secondary transports: " + s);
+							}
+						}
+						if(audio != null){
+							for(int a : audio){
+								Log.d(TAG, "Supported audio transports: " + a);
+							}
+						}
+						if(video != null){
+							for(int v : video){
+								Log.d(TAG, "Supported video transports: " + v);
+							}
+						}
+						handleEnableSecondaryTransport((byte) packet.getSessionId(), secondary, audio, video);

```

Comment:
Why is it necessary to call up to the SdlSession/IProtocolListener? At this point I believe the protocol layer should only retain what is available and what service should be connected. Looking at the code in SdlSession it seems it just calls right back to WiProProtocol.